packages/wallet-specs/src/store.ts

```diff
@@ -61,24 +74,35 @@ export class Store implements IStore {
   private _store: ChannelStore;
   private _privateKeys: Record<string, string>;
   private _nonces: Record<string, string> = {};
-
+  private _eventEmitter = new EventEmitter();
   protected _chain: IChain;
+  protected _messagingService: IMessageService;
 
   constructor(args?: Constructor) {
     const { store, privateKeys } = args || {};
     this._store = store || {};
     this._privateKeys = privateKeys || {};
 
+    // TODO: We probably shouldn't default to test implementations
     this._chain = args?.chain || new Chain();
-    // TODO: Bad form to call an async method in the constructor?
-    this._chain.initialize();
+    this._messagingService = args?.messagingService || messageService;
+
+    this._chain // TODO: Bad form to call an async method in the constructor?
+      .initialize();
   }
 
   public async getHoldings(channelId: string) {
     return await this._chain.getHoldings(channelId);
   }
-  public on(chainEventType, listener) {
-    return this._chain.on(chainEventType, listener);
+  public on(eventType: StoreEventType, listener: StoreEventListener) {
+    if (eventType === 'CHANNEL_UPDATED') {
```

Comment:
Could we `switch` on `eventType`, and return `unreachable(eventType)` as the default?